crates/biome_css_analyze/src/lint/nursery/no_invalid_position_at_import_rule.rs

```diff
@@ -0,0 +1,99 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{AnyCssRule, CssImportAtRule, CssRuleList};
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Disallow invalid position @import rules.
+    ///
+    /// Any @import rules must precede all other valid at-rules and style rules in a stylesheet (ignoring @charset and @layer), or else the @import rule is invalid.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// a {}
+    /// @import 'foo.css';
+    /// /** ↑
+    /// * This @import */
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// @import 'foo.css';
+    /// a {}
+    /// ```
+    ///
+    pub NoInvalidPositionAtImportRule {
+        version: "next",
+        name: "noInvalidPositionAtImportRule",
+        recommended: true,
+        sources: &[RuleSource::Stylelint("no-invalid-position-at-import-rule")],
+    }
+}
+
+impl Rule for NoInvalidPositionAtImportRule {
+    type Query = Ast<CssRuleList>;
+    type State = CssImportAtRule;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let node = ctx.query();
+        let mut is_invalid_position = false;
+
+        for rule in node {
+            let any_css_at_rule = match rule {
+                AnyCssRule::CssAtRule(item) => item.rule().ok(),
+                _ => None,
+            };
+
+            if let Some(any_css_at_rule) = any_css_at_rule {
+                // Ignore @charset, @layer
+                if any_css_at_rule.as_css_charset_at_rule().is_some() {
+                    continue;
+                }
+                if any_css_at_rule.as_css_layer_at_rule().is_some() {
+                    continue;
+                }
+
+                let import_rule = any_css_at_rule.as_css_import_at_rule().cloned();
+                if let Some(import_rule) = import_rule {
+                    if is_invalid_position {
+                        return Some(import_rule);
+                    }
+                } else {
+                    is_invalid_position = true;
+                }
+            } else {
+                is_invalid_position = true;
+            }
+        }
+
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, node: &Self::State) -> Option<RuleDiagnostic> {
+        //
+        // Read our guidelines to write great diagnostics:
+        // https://docs.rs/biome_analyze/latest/biome_analyze/#what-a-rule-should-say-to-the-user
+        //
+        let span = node.range();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                span,
+                markup! {
+                    "Unexpected import position."
+                },
+            )
+            .note(markup! {
+                    "Consider moving import position."
```

Comment:
Can you move this at the end? Reminder to check our rule pillars https://biomejs.dev/linter/#rule-pillars

Reply:
fix: https://github.com/biomejs/biome/pull/2717/commits/c6f45ab2cbb9b59801a65d71ad685069b0fa7355